### PR TITLE
SetPrivacy Route now handles all things privacy

### PR DESCRIPTION
**Pull request by zakhap**
## Description

- Removed privacy settings from `/editThread` so that only the user can use it (no admin editing threads) and place all privacy setting modifications in new route `/setPrivacy`.
- Added tests for `/setPrivacy` and removed old privacy setting tests from `/editThread` tests. 
- Updated the UI components for 'Lock Thread' and 'Make Public' (for private threads) to use the `setPrivacy` method I added to the `Threads Controller`. 

## Motivation and Context

No one but the author should use the editThread route, so another route and controller method was needed! 

## How has this been tested?

I wrote tests and played with the UI for both cases: toggling locked (readonly) on/off + setting a thread private in psql and making it public again via UI button.

## Have proper tags been added (for bug, enhancement, breaking change)?
- [x] yes

## Does this PR affect any server routes?
- [x] yes, and they are tested: 96.97% 💪

### CODE REVIEW

File: server/routes/setPrivacy.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      where: { id: thread_id, },
+      include: [ models.Address, models.OffchainAttachment, { model: models.OffchainTag, as: 'tag' } ],
+    })
+
+    return res.json({ status: 'Success', result: finalThread.toJSON() });
+  } catch (e) {
+    return next(new Error(e));
+  }
+
+  // Todo: dispatch notifications conditional on a new mention
```
Comment (raykyri) [inline server/routes/setPrivacy.ts:58]: Remove

File: client/scripts/controllers/server/threads.ts [hunk 2]
```diff
@@ -160,6 +154,30 @@ class ThreadsController {
     });
   }
 
+  public async setPrivacy({ threadId, privacy, readOnly }: { threadId: Number; privacy: boolean; readOnly: boolean; }) {
```
Comment (raykyri) [inline client/scripts/controllers/server/threads.ts:157]: couldn't these arguments just be specified as `{ threadId: Number; privacy: boolean; readOnly: boolean; }`?

File: server/routes/setPrivacy.ts (hunk 1 above)
Comment (zakhap) [reply server/routes/setPrivacy.ts]: fixed!

File: client/scripts/controllers/server/threads.ts (hunk 2 above)
Comment (zakhap) [reply client/scripts/controllers/server/threads.ts:157]: The typescript compiler wasn't happy with me typing the values while simultaneously destructuring. I opted for a similar, but less verbose solution.

**Pull request merged**